modules/java8/manifests/init.pp

```diff
@@ -19,19 +20,28 @@
   $hash          = '0da788060d494f5095bf8624735fa2f1',
   $load_dir      = "/tmp/",
   $arch_bit      = $java8::params::arch_bit,
+  $local_repo    = 'false',
+  $local_source  = undef,
 ) inherits java8::params {
 
  # validate java Standard Edition to download
   if $java_se !~ /(jre|jdk)/ {
     fail('Java SE must be either jre or jdk.')
   }
-
-  $pkg       = "${java_se}-8u${version_major}-${arch_bit}.rpm"
-  $source    = "${cookie} ${oracle_url}8u${version_major}-${version_minor}/${hash}/$pkg"
-  $pkg_path  = "${load_dir}${pkg}"
-  $java_path = "/usr/java/${java_se}1.8.0_${version_major}"
-  $env_filepath = "/etc/profile.d/app.sh"
-
+  if $local_repo == 'true' {
+    $pkg       = "${java_se}-8u${version_major}-${arch_bit}.rpm"
+    $source    = "${local_source}/${pkg}"
+    $pkg_path  = "${load_dir}${pkg}"
+    $java_path = "/usr/java/${java_se}1.8.0_${version_major}"
+    $env_filepath = "/etc/profile.d/app.sh"
+  }
+  else {
+    $pkg       = "${java_se}-8u${version_major}-${arch_bit}.rpm"
+    $source    = "${cookie} ${oracle_url}8u${version_major}-${version_minor}/${hash}/$pkg"
+    $pkg_path  = "${load_dir}${pkg}"
+    $java_path = "/usr/java/${java_se}1.8.0_${version_major}"
+    $env_filepath = "/etc/profile.d/app.sh"
+  }
```

Comment:
I would change only $source variable by condition to avoid repeating code.
Also, it could be more readable.